crates/cairo-lint-core/src/lints/collapsible_if_else.rs

```diff
@@ -0,0 +1,48 @@
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_syntax::node::ast::{Expr, ElseClause, BlockOrIf, ExprBlock, Statement};
+use cairo_lang_syntax::node::db::SyntaxGroup;
+use cairo_lang_syntax::node::{TypedStablePtr, TypedSyntaxNode};
+
+pub const COLLAPSIBLE_IF_ELSE: &str = "Consider using else if instead of else { if ... }";
+
+pub fn is_first_statement_if(db: &dyn SyntaxGroup, block_expr: &ExprBlock) -> bool {
+    // Get the list of statements from the block expression
+    let statements: Vec<Statement> = block_expr.statements(db).elements(db);
+
+    // Check if the first statement is an `if` statement
+    if let Some(first_statement) = statements.first() {
+
+        if let Statement::Expr(statement_expr) = first_statement {
+            let first_statement_expr = statement_expr.expr(db);
+            if let Expr::If(_) = first_statement_expr {
+                true
+            } else {
+                false
+            }
+        } else {
+            false
+        }
+    } else {
+        false
+    }
```

Comment:
too many indents here group the ifs together like that
```suggestion
    if let Some(first_statement) = statements.first() && let Statement::Expr(statement_expr) = first_statement && let Expr::If(_) = first_statement_expr = statement_expr.expr(db) {
    true
    } else {
        false
    }
```

Reply:
That suggestion gave some issues, but I improved the code in another way 😊